docs/rep/controller/openapi.yaml

```diff
@@ -0,0 +1,298 @@
+    post:
+      tags:
+        - Workloads
+      description: Create a new workload
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/WorkloadCreate'
+      responses:
+        '200':
+          description: Successful Response
+  /api/v0/workloads.delete:
+    post:
+      tags:
+        - Workloads
+      description: Delete a workload
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/WorkloadId'
+            
+      responses:
+        '200':
+          description: Successful Response
+  /api/v0/tenants.list:
+    get:
+      tags:
+        - Tenants
+      description: List all deployment
+      parameters:
+        - required: false
+          schema:
+            title: limit
+            minimum: 1
+            type: integer
+          name: limit
+          in: query
+      responses:
+        '200':
+          description: OK
+          content:
+            text/plain:
+              schema:
+                 $ref: '#/components/schemas/TenantList'
+  /api/v0/tenants.create:
+    post:
+      tags:
+        - Tenants
+      description: List all deployment
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Tenant'
+      responses:
+        '200':
+          description: Successful Response
+  /api/v0/tenants.delete:
+    post:
+      tags:
+        - Tenants
+      description: Delete a tenant
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/TenantDelete'
+      responses:
+        '200':
+          description: Successful Response
+  /api/v1/instances.list:
+    get:
+      tags:
+        - Instances
+      description: List all instances
+      parameters:
+        - required: false
+          schema:
+            title: limit
+            minimum: 1
+            type: integer
+          name: limit
+          in: query
+      responses:
+        '200':
+          description: OK
+          content:
+            text/plain:
+              schema:
+                 $ref: '#/components/schemas/InstanceList'
+          
+  /api/v1/instances.create:
+    post:
+      tags:
+        - Instances
+      description: Create a new instance
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/InstanceCreate'
+      responses:
+        '200':
+          description: Successful Response
+  /api/v0/instance.delete:
+    post:
+      tags:
+        - Instances
+      description: Delete an instance
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/InstanceId'
+      responses:
+        '200':
+          description: Successful Response
+        
+components:
```

Comment:
You should have 6 component schemes:
1. `TenantDefinition` (Used for creation) and `Tenant` (Used for listing)
2. `WorkloadDefinition` and `Workload`
3. `InstanceDefinition` and `Instance`